handlers/main.yml

```diff
@@ -36,3 +28,22 @@
   when:
     - openhpc_slurm_service_started | bool
     - openhpc_slurm_control_host in play_hosts
+  listen: Restart all slurm daemons
+
+- name: Mark slurmd as needing restart
+  set_fact:
+    openhpc_slurm_restart: true
+  delegate_to: "{{ ansible_play_hosts | first }}" # just need a predictable location
```

Comment:
Is it not easier not just to set it on all nodes?

Reply:
I guess what I mean is:

set_face:
  openhpc_slurm_restart: openhpc_slurm_service_started | bool

But maybe you need this to run once, so you only reastart all daemons once? My brain is failing me.

Reply:
I don't think that works - the problem is that the "Restart all slurm daemons" "listen-handler" or whatever the name is is only called for the control node (cause that's the only one where the config has Changed) so these tasks only run for that node. Is there a way of setting a variable on all hosts?

Reply:
I've done this before:

https://github.com/stackhpc/ansible-role-openhpc/commit/9d57521c58e90ca2df4f2614c22202e169ac77c5#diff-501ac8dd30adfb56d66b0c35b70bee6cb7b2eff2cbc0ef1e65215cf24ed417f4R27

But I think your method is probably cleaner.